hedera-node/src/main/java/com/hedera/services/config/NetworkInfo.java

```diff
@@ -47,8 +48,22 @@ public ByteString ledgerId() {
 			 *   0x02 -> previewnet
 			 *   0x03 -> other dev or preprod networks
 			 */
-			ledgerId = ByteString.copyFromUtf8(properties.getStringProperty("ledger.id"));
+			ledgerId = rationalize(properties.getStringProperty("ledger.id"));
 		}
 		return ledgerId;
 	}
+
+	private ByteString rationalize(String ledgerProperty) {
+		if (!ledgerProperty.startsWith("0x")) {
```

Comment:
Can't we just wrap a call to `com.swirlds.common.CommonUtils.unhex(ledgerProperty)` in a `try-catch (IllegalArgumentException)` and get the same result?

Reply:
Makes sense. But `unhex()` still needs the **`0x`**  to be missing from the string when it parses the big-endian representation.